app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,93 @@
+require './config/environment'
+
+class TweetController < ApplicationController
+
+  get '/tweets' do
+    if session[:user_id]
+      @user = User.find_by(session[:user_id])
+      @tweets = Tweet.all
+      #binding.pry
+      erb :tweets
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :new
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+      if logged_in?
+        if params[:content] == ""
+          redirect to "/tweets/new"
+        else
+          @tweet = current_user.tweets.build(content: params[:content])
+          if @tweet.save
+            redirect to "/tweets/#{@tweet.id}"
+          else
+            redirect to "/tweets/new"
+          end
+        end
+      else
+        redirect to '/login'
+      end
+    end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by(params[:id])
+      erb :show
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    #binding.pry
+    if logged_in?
+      @tweet = Tweet.find_by(params[:id])
+      erb :edit
+    else
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if logged_in?
+      if params[:content] == ""
+        redirect to "/tweets/#{params[:id]}/edit"
+      else
+        @tweet = Tweet.find_by_id(params[:id])
+        if @tweet && @tweet.user == current_user
+          if @tweet.update(content: params[:content])
+            redirect to "/tweets/#{@tweet.id}"
+          else
+            redirect to "/tweets/#{@tweet.id}/edit"
+          end
+        else
+          redirect to '/tweets'
+        end
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  delete '/tweets/:id/delete' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet && @tweet.user == current_user
```

Comment:
Same comment as patch `tweets/:id`: I don't see a lot of students checking the tweet's user id. Good job!